keps/sig-api-machinery/3157-watch-list/README.md

```diff
@@ -825,7 +836,8 @@ NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
 The expected behavior of the feature will be restored.
 
 ###### Are there any tests for feature enablement/disablement?
-No.
+Yes. There are tests that explicitly enable this feature for kube-apiserver and 
```

Comment:
These are not enablement tests - these are feature tests.
Enablement/disbalement tests are tests that switch the FG in the middle of the test.


On the server side, this feature is purely in-memory so no enablement/disablement tests are needed.
That said, we should create a test for the reflector that enables/disables the feature on the server side in the middle of its runtime.

Reply:
Do you have an example of such a test ?

As of today we have https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/watchlist.go#L38 which implicitly sets the FG on the server side.

Reply:
> That said, we should create a test for the reflector that enables/disables the feature on the server side in the middle of its runtime.

The reflector cannot enable/disable the feature on the server. The feature can be disabled by an FG that requires restarting the server. 

The same applies to the reflector/kcm, disabling requires passing an FG and restarting the process.

But I think that what we could do would be to extend the test I linked above (or write a new one actually) to:
1. use the new mode to get data
2. then disable the env var
3. then use the old mode to get data

Reply:
1. Reflector has the `UseWatchList` field to configure whether the feature should be used - changing that doesn't require component restart

2. You can disable the feature on kube-apiserver when the client is still running. That would actually be a very useful test to have as an integration tests [we don't have a good framework for it in e2e tests].

Reply:
I was asking for a link to an example test because I think that it is not possible to change FG on a running kas. 

This is because kas is run in a separate process (and changing another process's memory might be tricky 😅) and it doesn't react to the changes in the original flags that were passed to the server. Unless we could run more than one server.

I'm not sure if I understand what test we would like to have in this section. 

I think that we would like to test if a client can talk to servers with the FG enabled/disabled when it loses connectivity to a server. Is that correct ? In that case maybe it would be easier to test such a behaviour using unit tests on the client-side. (we could extend https://github.com/kubernetes/kubernetes/blob/04668c00432cbd552b08eb94f829643facbd2061/staging/src/k8s.io/client-go/tools/cache/reflector_watchlist_test.go#L111)

Reply:
came up with https://github.com/kubernetes/kubernetes/pull/120971, ptal